pkg/crd/markers/validation.go

```diff
@@ -284,3 +294,55 @@ func (m Nullable) ApplyToSchema(schema *v1beta1.JSONSchemaProps) error {
 	schema.Nullable = true
 	return nil
 }
+
+// Defaults are only valid CRDs created with the v1 API
+func (m Default) ApplyToSchema(schema *v1beta1.JSONSchemaProps) error {
+	// Convert versioned schema to unversioned form
+	marshalledSchema, err := json.Marshal(schema)
+	if err != nil {
+		return err
+	}
+	unversionedSchema := &apiext.JSONSchemaProps{}
+	err = json.Unmarshal(marshalledSchema, unversionedSchema)
+	if err != nil {
+		return err
+	}
+
+	// Initialize validator for schema
+	crValidation := &apiext.CustomResourceValidation{
+		OpenAPIV3Schema: unversionedSchema,
+	}
+	validator, _, err := validation.NewSchemaValidator(crValidation)
+	if err != nil {
+		return err
+	}
```

Comment:
if you really want to validate the value here, you can just unmarshal the schema bytes into a go-openapi spec and call go-openapi's validation directly, without accessing apiextensions-apiserver internals.

Reply:
I didn't realize these exported methods were intended to be internal. How do I determine whether any given functionality provided by a dependency is fair game for use?

Reply:
*sigh* I guess the use of unversioned API types would be my first clue.

Reply:
They might go away any time. If we finally move internal packages to `internal`, there won't be a replacement.

Reply:
I am fine with this now.